base/broadcast.jl

```diff
@@ -1377,4 +1380,35 @@ function Base.show(io::IO, op::BroadcastFunction)
 end
 Base.show(io::IO, ::MIME"text/plain", op::BroadcastFunction) = show(io, op)
 
+
+# Fill Broadcasting
+function broadcasted(::DefaultArrayStyle{N}, op, r::AbstractFill{T,N}) where {T,N}
+    return Fill(op(getindex_value(r)), axes(r))
+end
+
+function broadcasted(::DefaultArrayStyle, op, a::AbstractFill, b::AbstractFill)
+    val = op(getindex_value(a), getindex_value(b))
+    return Fill(val, broadcast_shape(axes(a), axes(b)))
+end
+
+# Need to prevent array-valued fills from broadcasting over entry
+_broadcast_getindex_value(a::AbstractFill{<:Number}) = getindex_value(a)
+_broadcast_getindex_value(a::AbstractFill) = Ref(getindex_value(a))
+
+function broadcasted(::DefaultArrayStyle{1}, ::typeof(*), a::AbstractFill, b::AbstractRange)
+    broadcast_shape(axes(a), axes(b)) == axes(b) || throw(ArgumentError("Cannot broadcast $a and $b. Convert $b to a Vector first."))
+    return broadcasted(*, _broadcast_getindex_value(a), b)
+end
+
+function broadcasted(::DefaultArrayStyle{1}, ::typeof(*), a::AbstractRange, b::AbstractFill)
+    broadcast_shape(axes(a), axes(b)) == axes(a) || throw(ArgumentError("Cannot broadcast $a and $b. Convert $b to a Vector first."))
```

Comment:
This was carried over from FillArrays.jl but I think was a temporary restriction. I'm not entirely sure this restriction should be kept.

Reply:
Should I just delete that line altogether?

Reply:
The check is necessary because the function may be incorrect  in the "degenerate" case where `a` is length `1` and so the shape comes from `b`, not `a`. Perhaps it would be better to be rewritten as
```julia
if length(a) == 1 
   broadcasted(*, a[1], b)
else
   broadcasted(*, a, _broadcast_getindex_value(b))
end
```
but this would not be type stable. 

In any case some thought should be put into this.